### DISCUSSION TITLE
New tutorial "Set up an Xray-core VLESS proxy to access blocked websites and apps"

### DISCUSSION THREAD

**Discussion by wpdevelopment11** [XTLS/Xray-core#3981] [Show and tell]
I have written a tutorial.

["Set up an Xray-core VLESS proxy to access blocked websites and apps"](https://github.com/wpdevelopment11/xray-tutorial)

In this tutorial we will configure an Xray proxy server which will be used to circumvent blocking of websites and apps.

I hope that it will be useful for somebody.

**Comment by wpdevelopment11** [comment]
Updated tutorial. Added instructions for the [Cloudflare proxy](https://github.com/wpdevelopment11/xray-tutorial?tab=readme-ov-file#step-4---cloudflare-proxy-optional).

**Comment by wpdevelopment11** [comment]
Added instructions for [ad blocking and website blocking in general](https://github.com/wpdevelopment11/xray-tutorial?tab=readme-ov-file#step-5---blocking-ads-optional).

BTW:
The latest version of Xray (25.3.6) can't connect to the server behind the Cloudflare proxy using WebSocket (ws) transport. Older version of Xray (24.11.11) works just fine. ~~Most likely it's a bug in Xray.~~ The cause is unknown, now it works.

I get the following errors on the client side:

```
transport/internet/websocket: failed to dial to mydomain.com:443 > dial tcp: lookup mydomain.com: operation was canceled
wsarecv: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
```

**Comment by wpdevelopment11** [comment]
[Updated tutorial](https://github.com/wpdevelopment11/xray-tutorial).

**Reply by algo7** [reply]
Nice tutorial 👍.  Can you also do one for Wireguard inbound (relatively new) on Linux client? Because having to configure SOCKS for each app is very time-consuming. The system-wide SOCKS settings solved some of the problems, but there are apps that just dont support it. Having some "tunnel" that wraps the traffic at layer 2-4  can avoid all these issues.

**Reply by wpdevelopment11** [reply]
Thanks. I will most likely not do it, cause I don't use WireGuard myself. If you want to route all your traffic through Xray you can also take a look into [transparent proxy tutorial](https://xtls.github.io/document/level-2/tproxy.html) with a help of machine translation.

**Comment by wpdevelopment11** [comment]
Updated tutorial.
Described [XTLS Vision and Reality configuration](https://github.com/wpdevelopment11/xray-tutorial#step-5---configuring-xtls-vision-and-reality).

**Comment by wpdevelopment11** [comment]
Updated tutorial.

Added the instructions for [Reality  with a `grpc` transport](https://github.com/wpdevelopment11/xray-tutorial#step-54---grpc-transport) which is useful when Reality with `raw`/`tcp` transport is blocked.

This tutorial mirrors the evolution of blocking methods in my region. You may need to adapt it for your own needs.

**Comment by wpdevelopment11** [comment]
Updated [the tutorial](https://github.com/wpdevelopment11/xray-tutorial). Added an XHTTP configuration.
